### PR TITLE
Linting on Handlebars :crossed_swords:

### PR DESCRIPTION
**Pull request by jonahkaye**
Refs: #[1474](https://github.com/metriport/metriport-internal/issues/1474)

### Description

- ember-lint + postprocessing since linter config file isnt working 
- basically ember-template-lint (which is the only handlebars linter that worked I found) has a known issue: https://github.com/ember-template-lint/ember-template-lint/issues/486 which doesn't let you override handlebars-partials in the config. So basically, we have to post-process the linting output for bugs that dont matter to us

### Testing

- Local
  - [x] linting. I tried to introduce hella bugs and we catch em all baby 

Check each PR.

### Release Plan

- [ ] Merge this

### CODE REVIEW

File: packages/fhir-converter/src/templates/cda/Utils/GenerateId.hbs [hunk 1]
```diff
@@ -8,7 +8,7 @@
                 "{{generateUUID (toString FirstId.root)}}"
             {{else}}
                 {{!--Generate using obj if no root or id is available--}}
-                "{{generateUUID (toJsonString ..)}}"
+                "{{generateUUID (toJsonString ../obj)}}"
```
Comment (jonahkaye) [inline packages/fhir-converter/src/templates/cda/Utils/GenerateId.hbs:11]: from handlebars lint

**Review by jonahkaye — COMMENTED** [review]
Example of the kind of bug we accidentally shipped to prod. Missing a single brace, you might not pick it up until too late or forget to test on all 10k. Will stop us from shipping bugs

```
jonahkaye@Jonahs-MBP metriport % git commit -m "feat(bad): dont let"

> metriport@1.0.0 check-secrets
> docker run --rm -v $(pwd):/path zricethezav/gitleaks:v8.17.0 protect --source='/path' --staged --no-banner -v

8:54PM INF 1 commits scanned.
8:54PM INF scan completed in 60.2ms
8:54PM INF no leaks found

> metriport@1.0.0 lint-staged
> npx lint-staged

→ No staged files match any configured task.

> metriport@1.0.0 full-hbs-lint
> cd packages/fhir-converter && ./full-hbs-lint.sh


> fhirconverter@2.1.1 lint:hbs
> ember-template-lint src/templates/cda > lint-results.txt


> fhirconverter@2.1.1 postprocess-hbs-lint
> node postprocess-hbs-lint.js

Linting issues found:
src/templates/cda/Resources/Location.hbs
  -:-  error  Parse error on line 10:
...            {{/each}        ],       
----------------------^
Expecting 'CLOSE_RAW_BLOCK', 'CLOSE', 'CLOSE_UNESCAPED', 'OPEN_SEXPR', 'CLOSE_SEXPR', 'ID', 'OPEN_BLOCK_PARAMS', 'STRING', 'NUMBER', 'BOOLEAN', 'UNDEFINED', 'NULL', 'DATA', 'SEP', got 'INVALID'  undefined
husky - pre-commit hook exited with code 1 (error)
```

**Review by jonahkaye — COMMENTED** [review]
And when there are no errors: 

```
git commit -m "feat(linting): works when there are no errors" -m "Refs: #1474"

> metriport@1.0.0 check-secrets
> docker run --rm -v $(pwd):/path zricethezav/gitleaks:v8.17.0 protect --source='/path' --staged --no-banner -v

10:06PM INF 1 commits scanned.
10:06PM INF scan completed in 56.6ms
10:06PM INF no leaks found

> metriport@1.0.0 lint-staged
> npx lint-staged

→ No staged files match any configured task.

> metriport@1.0.0 full-hbs-lint
> cd packages/fhir-converter && ./full-hbs-lint.sh


> fhirconverter@2.1.1 lint:hbs
> ember-template-lint src/templates/cda > lint-results.txt

→ No linting issues found
[1474-add-handlebars-linting a824c491] feat(linting): works when there are no errors
 2 files changed, 5 insertions(+), 7 deletions(-)
```

File: packages/fhir-converter/full-hbs-lint.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+LINT_RESULTS=$(node postprocess-hbs-lint.js)
+
+# Check if the output is not empty
+if [ ! -z "$LINT_RESULTS" ]; then
+  echo "Linting issues found:"
+  echo "$LINT_RESULTS"
+  rm lint-results.txt
+  exit 1
+else
+  echo -e "\033[34m→\033[0m No linting issues found"
```
Comment (jonahkaye) [inline packages/fhir-converter/full-hbs-lint.sh:12]: blue arrow

**Review by Goncharo — APPROVED** [review]
🤰 🤰 🤰 🤰 

only nit to think about later is to ideally have the postprocess linting be `ts` code, not `js` if possible

**Pull request merged**